InformationScripting/src/query_prompt/QueryPromptMode.cpp

```diff
@@ -68,36 +67,22 @@ void QueryPromptMode::onEnterKeyPress(Qt::KeyboardModifiers)
 		// Note a QueryExecutor should always be allocated with new, it is self destroying:
 		auto executor = new QueryExecutor();
 		QueryBuilder builder{node, executor};
-		executor->addQuery(builder.visit(queryNode));
-		result = executor->execute();
-	}
-	else
-		result = new Interaction::CommandResult(new Interaction::CommandError("Queries only work on nodes"));
-
-	if ( result->code() == Interaction::CommandResult::OK) Interaction::Prompt::hide();
-	else
-	{
-		QList<Visualization::Item*> errorItems;
-
-		for (auto& error : result->errors() )
+		try
 		{
-			if (error->visualization() == nullptr)
-			{
-				auto vis = new Interaction::TextAndDescription(nullptr,
-											Interaction::TextAndDescription::itemStyles().get("command-prompt-error"));
-				vis->setContents(error->message(), error->resolutionTips().join(" OR "));
-				errorItems.append(vis);
-			}
-			else
-			{
-				//Extract the visualization
-				errorItems.append(error->visualization());
-				error->setVisualization(nullptr);
-			}
+			executor->addQuery(builder.visit(queryNode));
+			errors = executor->execute();
+		}
+		catch (const QueryParsingException& parsingException)
+		{
+			errors = {parsingException.message()};
+			SAFE_DELETE(executor);
```

Comment:
Hm true, but since the exception would be thrown in the first statement ```builder.visit(..)``` the execute, which would destroy the executor is never executed, so we have to destroy it here.